sympy/solvers/inequalities.py

```diff
@@ -481,11 +482,25 @@ def valid(x):
         if valid(pt):
             sol_sets.append(Interval(start, end, True, True))
 
-        rv = Union(*sol_sets).subs(gen, _gen)
+        elif solns == []: # if no solutions for equation e
+            nsol_sets = _inequalities_without_roots(expr,gen)
 
+        cv = Intersection(*nsol_sets).subs(gen,_gen)
+        rv = Union(*sol_sets).subs(gen, _gen)
+        rv = Union(rv, cv)
     return rv if not relational else rv.as_relational(_gen)
 
 
+def _inequalities_without_roots(expr, gen):
+    n_sol_sets = []
+    e = expr.gts - expr.lts
+    if e.is_Add:
+        for subexpr in e.args:
+            n_sol_sets.append(solve_univariate_inequality(subexpr > 0, gen, relational = False))
```

Comment:
- the expression `e = expr.gts - expr.lts` makes sure that `subexpr>0`.

- Also `e.is_Add = True` , (line 495) makes sure that `e` subfunctions are being added (not multiplied or divided)